src/panels/config/entities/entity-registry-settings.ts

```diff
@@ -719,6 +722,15 @@ export class EntityRegistrySettings extends SubscribeMixin(LitElement) {
     showOptionsFlowDialog(this, this._helperConfigEntry!);
   }
 
+  private _switchAsDomainsSorted = memoizeOne((domains: string[]) =>
+    domains
+      .map((entry) => ({
+        domain: entry,
+        translation: domainToName(this.hass.localize, entry),
```

Comment:
`localize` should be part of the params of the memoized function, so if the language/translation changes, the label also changes